File: APCS202101_飛黃騰達(20%).cpp
#include <bits/stdc++.h>

using namespace std;

#define N 50010
typedef long long ll;

int fun(int n, int m, int k)
{
    if (k == 0)
        return 0;
    return (fun(n - 1, m, k - 1) + m) % n;
}

int main()
{
    // freopen("P_3_8_5.in", "r", stdin);
    int n, x, y;
    cin >> n;
    vector<vector<int>> mat(101, vector<int>(101, 0));
    for (int i = 0; i < n; ++i)
    {
        cin >> x >> y;
        mat[x][y] = 1;
    }
    for (int i = 1; i <= 100; ++i)
    {
        for (int j = 1; j <= 100; ++j)
            mat[i][j] += max(mat[i - 1][j], mat[i][j - 1]);
    }
    cout << mat[100][100];
    return 0;
}

File: APCS202101_飛黃騰達(20%).py
n = int(input())
mat = [[0]*101 for _ in range(101)]
for _ in range(n):
    x, y = map(int, input().split())
    mat[x][y] = 1
for r in range(1,101):
    for c in range(1,101):
        mat[r][c] += max(mat[r-1][c], mat[r][c-1])
print(mat[100][100])
